### PR TITLE
Error Summary component in storybook

### LINKED ISSUE #1636: Using lingui and storybook
Discussion on using lingui and storybook
https://github.com/lingui/js-lingui/issues/413

_Originally posted by @sastels in https://github.com/cds-snc/report-a-cybercrime/pull/1620#issuecomment-594585817_

### PR DESCRIPTION
**Pull request by amazingphilippe**
Fixes #1636 

# Description

Added ErrorSummary to Storybook docs. 
Also fixed using lingui in storybooks

# Any new packages installed?

Updated prettier from 1.19 to 2.0.2. Fixes pre-commit failures on .mdx files. 

# Required followup work

Prettier got more [opinionated](https://prettier.io/blog/2020/03/21/2.0.0.html#javascript-1) in js files. Expect some reformating when saving files. 

# Checklist:

- [x] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [x] I have added and needed tests for my changes (in particular for new screens)
- [x] I have added a comment to any confusing code
- [x] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

File: f2/.storybook/formDecorator.js [hunk 1]
```diff
@@ -1,14 +1,19 @@
 import React from 'react'
 import { Form } from 'react-final-form'
 
-const FormDecorator = storyFn => (
+const validate = () => {
+  const errors = { email: 'contactinfoForm.email.warning' }
+  return errors
+}
+
```
Comment (amazingphilippe) [inline f2/.storybook/formDecorator.js]: Added a validate function in formDecorator for storybook

File: f2/.storybook/themeDecorator.js [hunk 2]
```diff
@@ -3,13 +3,13 @@ import { ThemeProvider, CSSReset, Box } from '@chakra-ui/core'
...
 import { Global, css } from '@emotion/core'
 import { MemoryRouter } from 'react-router-dom'
-import { I18nProvider, useLingui } from '@lingui/react'
+import { I18nProvider } from '@lingui/react'
 import { i18n } from '@lingui/core'
-import en from '../src/locales/en.json'
+import { activate } from '../src/i18n.config'
+
+const ThemeDecorator = (storyFn) => {
+  activate('en')
```
Comment (amazingphilippe) [inline f2/.storybook/themeDecorator.js]: this is how we make lingui work in storybook

File: f2/src/components/ErrorSummary/ErrorSummary.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'
+import ThemeDecorator from '../../../.storybook/themeDecorator'
+import FormDecorator from '../../../.storybook/formDecorator'
+import { withKnobs, boolean, text } from '@storybook/addon-knobs'
+import { Field } from '../Field'
+import { Input } from '../input'
+import { TextArea } from '../text-area'
+import { ErrorSummary } from '.'
+import { Stack } from '@chakra-ui/core'
+import { Trans } from '@lingui/react'
```
Comment (amazingphilippe) [inline f2/src/components/ErrorSummary/ErrorSummary.stories.mdx]: import trans from lingui/react, not lingui/macro. Only in storybook. [See this reference](https://lingui.js.org/tutorials/react.html#macros-vs-components)

File: f2/src/forms/PrivacyConsentInfoForm.js [hunk 4]
```diff
@@ -57,15 +57,13 @@ export const PrivacyConsentInfoForm = props => {
             shouldWrapChildren
             spacing={6}
           >
-            {submitFailed && hasValidationErrors ? (
-              <ErrorSummary onSubmit={handleSubmit} errors={errors} />
-            ) : null}
+            {submitFailed && hasValidationErrors ? <ErrorSummary /> : null}
```
Comment (sastels) [inline f2/src/forms/PrivacyConsentInfoForm.js]: Why don’t you pass on submit and errors anymore?

Comment (amazingphilippe) [reply f2/src/forms/PrivacyConsentInfoForm.js]: It appeared that we did not need them from the beginning...

I've tried removing them and things just worked. Is it possible that ErrorSummary inherits props from the Form component?

**Pull request merged**